server/src/main/java/org/jetbrains/bsp/bazel/server/bloop/BspProjectExporter.java

```diff
@@ -0,0 +1,94 @@
+package org.jetbrains.bsp.bazel.server.bloop;
+
+import com.google.common.collect.Maps;
+import io.vavr.PartialFunction;
+import io.vavr.collection.HashMap;
+import io.vavr.collection.HashSet;
+import io.vavr.collection.Set;
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.jetbrains.bsp.bazel.server.sync.languages.LanguageData;
+import org.jetbrains.bsp.bazel.server.sync.languages.java.JavaModule;
+import org.jetbrains.bsp.bazel.server.sync.languages.scala.ScalaModule;
+import org.jetbrains.bsp.bazel.server.sync.model.Module;
+import org.jetbrains.bsp.bazel.server.sync.model.Project;
+
+class BspProjectExporter {
+  private static final Set<Path> IGNORED_SOURCES =
```

Comment:
i dont like coupling the implementation with the twitter codebase - we need to come up with a better mechanism

Reply:
yea, we'll definitely come up with something better in the future